### PR TITLE
feat(LINGUIST-201): Added profile and hobby related controller service and repository

### PR DESCRIPTION
**Pull request by elif-krvan**

### CODE REVIEW

**Review by can-aslan — CHANGES_REQUESTED** [review]
Looks very good, just left some minor comments 🚀

File: src/main/java/app/linguistai/bmvp/controller/ProfileController.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import lombok.AllArgsConstructor;
+
+@AllArgsConstructor
+@RestController
+@Validated
+@RequestMapping("profile")
+public class ProfileController {
+    private final ProfileService profileService;
+
+    @PutMapping(consumes = MediaType.APPLICATION_JSON_VALUE, path = "")
```
[the remark this answers is not in the document]
Comment (elif-krvan) [reply src/main/java/app/linguistai/bmvp/controller/ProfileController.java:30]: napim

File: src/main/java/app/linguistai/bmvp/controller/ProfileController.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+        try {
+            RUserProfile userProfile = profileService.updateUserProfile(email, profile);
+            return Response.create("Profile is updated", HttpStatus.OK, userProfile);
+        } catch (Exception e) {
+            return Response.create(ExceptionLogger.log(e), HttpStatus.BAD_REQUEST);
+        }        
+    }
+
+    @GetMapping("")
```
[the remark this answers is not in the document]
Comment (can-aslan) [reply src/main/java/app/linguistai/bmvp/controller/ProfileController.java:42]: `"ne"` gibi soru ekleri ayrı yazılır 🤓

File: src/main/java/app/linguistai/bmvp/model/profile/Hobby.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+@Data
+@Entity
+@Table(name = "hobby")
+@NoArgsConstructor
+public class Hobby {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "id", nullable = false)
+    private Integer id;
```
[the remark this answers is not in the document]
Comment (elif-krvan) [reply src/main/java/app/linguistai/bmvp/model/profile/Hobby.java:21]: They are just hobbies there is no need to make it too complicated

File: src/main/java/app/linguistai/bmvp/controller/ProfileController.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (can-aslan) [reply src/main/java/app/linguistai/bmvp/controller/ProfileController.java:30]: duzelt

File: src/main/java/app/linguistai/bmvp/model/profile/Hobby.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (can-aslan) [reply src/main/java/app/linguistai/bmvp/model/profile/Hobby.java:21]: Can we make it at least `Long`

[the remark this answers is not in the document]
Comment (elif-krvan) [reply src/main/java/app/linguistai/bmvp/model/profile/Hobby.java:21]: no

**Pull request merged**